akka-docs/src/main/paradox/typed/cluster-sharding-specification.md

```diff
@@ -0,0 +1,125 @@
+owning the entities will acknowledge the handoff as completed to the coordinator.
+Thereafter the coordinator will reply to requests for the location of
+the shard, thereby allocating a new home for the shard, and then buffered messages in the
+`ShardRegion` actors are delivered to the new location. This means that the state of the entities
+are not transferred or migrated. If the state of the entities are of importance it should be
+persistent (durable), e.g. with @ref:[Persistence](persistence.md) (or see @ref:[Classic Persistence](../persistence.md)), so that it can be recovered at the new
+location.
+
+The logic that decides which shards to rebalance is defined in a pluggable shard
+allocation strategy. The default implementation `ShardCoordinator.LeastShardAllocationStrategy`
+picks shards for handoff from the `ShardRegion` with most number of previously allocated shards.
+They will then be allocated to the `ShardRegion` with least number of previously allocated shards,
+i.e. new members in the cluster.
+
+For the `LeastShardAllocationStrategy` there is a configurable threshold (`rebalance-threshold`) of
+how large the difference must be to begin the rebalancing. The difference between number of shards in
+the region with most shards and the region with least shards must be greater than the `rebalance-threshold`
+for the rebalance to occur.
+
+A `rebalance-threshold` of 1 gives the best distribution and therefore typically the best choice.
+A higher threshold means that more shards can be rebalanced at the same time instead of one-by-one.
+That has the advantage that the rebalance process can be quicker but has the drawback that the
+the number of shards (and therefore load) between different nodes may be significantly different.
+
+### ShardCoordinator state
+
+The state of shard locations in the `ShardCoordinator` is persistent (durable) with
+@ref:[Distributed Data](distributed-data.md) (or see @ref:[Classic Distributed Data](../distributed-data.md))to survive failures. 
+
+When a crashed or
+unreachable coordinator node has been removed (via down) from the cluster a new `ShardCoordinator` singleton
+actor will take over and the state is recovered. During such a failure period shards
+with a known location are still available, while messages for new (unknown) shards
+are buffered until the new `ShardCoordinator` becomes available.
+
+### Message ordering
+
+As long as a sender uses the same `ShardRegion` actor to deliver messages to an entity
+actor the order of the messages is preserved. As long as the buffer limit is not reached
+messages are delivered on a best effort basis, with at-most once delivery semantics,
+in the same way as ordinary message sending. Reliable end-to-end messaging, with
+at-least-once semantics can be added by using `AtLeastOnceDelivery` in @ref:[Persistence](persistence.md) (or see @ref:[Classic Persistence](../persistence.md)).
```

Comment:
AtLeastOnceDelivery doesn't exist in Typed yet. We could add a section in the docs there and there refer to the issue? https://github.com/akka/akka/issues/20984